bin/pt-eustack-resolver

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env perl
+
+# This program is part of Percona Toolkit: http://www.percona.com/software/
+# See "COPYRIGHT, LICENSE, AND WARRANTY" at the end of this file for legal
+# notices and disclaimers.
+
+use strict;
+use Data::Dumper;
+
+my $mmap;
+
+sub load_mapping {
+  my ($pid)= @_;
+
+  if ($pid =~ /^[0-9]+$/) {
+    open F, '<', "/proc/$pid/maps"
+        or die "Failed to open /proc/$pid/maps: $!\n";
+  } else {
+    open F, '<', $pid
+        or die "Failed to open saved map file '$pid': $!\n";
+  }
+
+  my $arr= [];
+  while (<F>) {
+    next unless m/^([a-f0-9]+)-([a-f0-9]+) ..x. ([a-f0-9]+) [a-f0-9:]+ [0-9]+ +(.*)/;
+    push @$arr, { S => hex($1), E => hex($2), B => hex($3), F => $4 };
+  }
+  close F;
+  sort { $a->{S} <=> $b->{S} } @$arr;
+  $mmap= $arr;
+}
+
+my $syms= { };
+
+sub get_image {
+  my ($addr)= @_;
+  my $e;
+  for $e (@$mmap) {
+    next if $e->{E} <= $addr;
+    last if $e->{S} > $addr;
+    # Found, look up.  
+      return $e->{F};
+  }
+  return "";
+}
+
```

Comment:
sub get_image {
    my ($addr) = @_;

    # Ensure addr is defined
    die "Address is undefined" unless defined $addr;

    # Check if the global memory map is defined and is an array reference
    die "Global memory map is undefined or not an array reference"
        unless defined $memory_map and ref($memory_map) eq 'ARRAY';

    foreach my $entry (@$memory_map) {
        next if $entry->{E} <= $addr;
        last if $entry->{S} > $addr;
        # Found, return file
        return $entry->{F};
    }

    return "";
}

I would suggest trying something like this